lua/lualine.lua

```diff
@@ -148,34 +150,66 @@ end
 
 local function statusline(sections, is_focused)
   if M.theme ~= theme_set then
-    set_lualine_theme()
-  end
-  local status = {}
-  if sections.lualine_a then
-    local hl = highlight.format_highlight(is_focused, 'lualine_a')
-    table.insert(status, utils_component.draw_section(sections.lualine_a,  hl))
-  end
-  if sections.lualine_b then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_b,  hl))
-  end
-  if sections.lualine_c then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_c,  hl))
+    _G.lualine_set_theme()
   end
-  table.insert(status, "%=")
-  if sections.lualine_x then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_x,  hl))
-  end
-  if sections.lualine_y then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_y,  hl))
+  -- status_builder stores statusline without section_separators
+  local status_builder = {}
+  -- The sequence sections should maintain
+  local section_sequence = {'a', 'b', 'c', 'x', 'y', 'z'}
+
+  for _, section_name in ipairs(section_sequence) do
+    if sections['lualine_'..section_name] then
+      -- insert highlight+components of this section to status_builder
+      local section_highlight = highlight.format_highlight(is_focused,
+                   'lualine_'..section_name)
+			local section_data = utils_component.draw_section(sections['lualine_'..section_name], section_highlight)
+			if #section_data > 0 then
+				table.insert(status_builder, {section_name, section_data})
+			end
+    end
   end
-  if sections.lualine_z then
-    local hl = highlight.format_highlight(is_focused, 'lualine_a')
-    table.insert(status, utils_component.draw_section(sections.lualine_z,  hl))
+
+  -- Actual statusline
+  local status = {}
+  local half_passed = false
+  for i=1,#status_builder do
+    -- midsection divider
+    if not half_passed and status_builder[i][1] > 'c' then
+      table.insert(status, highlight.format_highlight(is_focused, 'lualine_c') .. '%=')
+      half_passed = true
+    end
+    -- provide section_separators when statusline is in focus
+    if is_focused then
+      -- component separator needs to have fg = current_section.bg
+      -- and bg = adjacent_section.bg
+      local previous_section = status_builder[i-1] and status_builder[i-1] or {}
```

Comment:
I still don't get what's this doing. I'm guessing
```
previous_section = status_builder[i-1] if status_builder[i-1] else {}
```
right?

Reply:
Yes

Reply:
Okay, can  you change it to a and if statement then? It will be more intuitive

Reply:
simplified